hudi-client/hudi-client-common/src/main/java/org/apache/hudi/table/HoodieTimelineArchiveLog.java

```diff
@@ -165,13 +167,17 @@ public boolean archiveIfRequired(HoodieEngineContext context) throws IOException
     HoodieTimeline commitTimeline = table.getCompletedCommitsTimeline();
     Option<HoodieInstant> oldestPendingCompactionInstant =
         table.getActiveTimeline().filterPendingCompactionTimeline().firstInstant();
+    Option<HoodieInstant> oldestInflightInstant =
+        table.getActiveTimeline()
+            .getTimelineOfActions(CollectionUtils.createSet(HoodieTimeline.COMMIT_ACTION, HoodieTimeline.DELTA_COMMIT_ACTION))
```

Comment:
is n't there an existing method to get this? Also should we include `replacecommit` here as well?  I understand the issue with compactions, since the retry will attempt to reuse the same instantTime. is that why you only have commit/deltacommit here?

Reply:
Yes, the retry of both will use the same instant right now. All of the lazy cleaning is only done for commits/deltacommits since there is contention when multiple writers start running. Both compaction and clustering will continue to do inline rollbacks since both these operations are bound to happen once scheduled. If the compaction or clustering DON'T run, the failed writes will keep lying around. Basically, once a compaction/clustering is scheduled, we ensure no other writer can modify the same set of files, there is no contention for the failed writes and hence the cleanup happens inline. If they don't happen inline and depend on the LAZY cleaning, then since we re-use the same instant, we cannot guarantee correctness.

Reply:
makes sense to have the lazy cleaning for just the true write actiosn.